landlab/components/weathering/exponential_weathering.py

```diff
@@ -109,17 +109,23 @@ def __init__(
         ----------
         grid: ModelGrid
             Landlab ModelGrid object
-        soil_production__maximum_rate : float
+        soil_production__maximum_rate : float, array of float
             Maximum weathering rate for bare bedrock
-        soil_production__decay_depth : float
+        soil_production__decay_depth : float, array of float
             Characteristic weathering depth
         """
         super().__init__(grid)
 
         # Store grid and parameters
+        self._w0 = np.broadcast_to(soil_production__maximum_rate, grid.number_of_nodes)
+        self._wstar = np.broadcast_to(
+            soil_production__decay_depth, grid.number_of_nodes
+        )
 
-        self._wstar = soil_production__decay_depth
-        self._w0 = soil_production__maximum_rate
+        if np.any(self._w0 <= 0):
+            raise ValueError("Maximum weathering rate must be positive.")
+        if np.any(self._wstar <= 0):
+            raise ValueError("Maximum decay depth must be positive.")
```

Comment:
You don't need these checks if you use the setters for *_w0* and *_wstar*.

Reply:
👍